yeonjae/programmers-1/콜라츠 추측.js

```diff
@@ -0,0 +1,17 @@
+function solution(num) {
+  let answer = 0;
+
+  for (let i = 0; i < 500; i++) {
+    if (num === 1) return (answer = i);
+    num = num % 2 == 0 ? num / 2 : num * 3 + 1;
+  }
+
+  return (answer = -1);
+}
```

Comment:
```suggestion
function solution(num) {
  for (let i = 0; i < 500; i++) {
    if (num === 1) return i;
    num = num % 2 == 0 ? num / 2 : num * 3 + 1;
  }
  return -1;
}
```
- 잘푸셨습니다  👍  하연님과 같은 피드백인데 굳이 `answer` 에다가 값을 할당하고 `return` 할 필요없이 곧바로 리턴하는게 더 좋아보입니다
- 위와 같이 할 경우 불 필요한 answer라는 변수도 필요 없어 질 것 같습니다 🙆‍♂️
- 아래 코드는 while문을 활용한 코드입니다. 참고해보세요~!! 🙇‍♂️
```js
function solution(num) {
  let result = 0;

  while (num > 1) {
    if (result > 500) return -1;

    num = num % 2 === 0 ? num / 2 : num * 3 + 1;
    result++;
  }
  return result;
}
```

Reply:
피드백 확인했습니다!!! @ssi02014